diff --git a/dist/compiled/react-dom/cjs/react-dom-client.development.js b/dist/compiled/react-dom/cjs/react-dom-client.development.js
index 4376913076151ad78dcc4059ad4152fad25478da..acfe822ba423aef22832121e1b696212b36fb8d1 100644
--- a/dist/compiled/react-dom/cjs/react-dom-client.development.js
+++ b/dist/compiled/react-dom/cjs/react-dom-client.development.js
@@ -15776,6 +15776,14 @@
           break;
         case 5:
           resetHooksOnUnwind(unitOfWork);
+          var fiber = unitOfWork;
+          fiber === hydrationParentFiber &&
+            (isHydrating
+              ? (popToNextHostParent(fiber),
+                5 === fiber.tag &&
+                  null != fiber.stateNode &&
+                  (nextHydratableInstance = fiber.stateNode))
+              : (popToNextHostParent(fiber), (isHydrating = !0)));
         default:
           unwindInterruptedWork(current, unitOfWork),
             (unitOfWork = workInProgress =
diff --git a/dist/compiled/react-dom/cjs/react-dom-client.production.js b/dist/compiled/react-dom/cjs/react-dom-client.production.js
index 755076a5cf15273f024c18fdc6a8805f1e36860b..9f656d03b19857cc91150746382d28c2451031fb 100644
--- a/dist/compiled/react-dom/cjs/react-dom-client.production.js
+++ b/dist/compiled/react-dom/cjs/react-dom-client.production.js
@@ -11210,6 +11210,14 @@ function replaySuspendedUnitOfWork(unitOfWork) {
       break;
     case 5:
       resetHooksOnUnwind(next);
+      var fiber = next;
+      fiber === hydrationParentFiber &&
+        (isHydrating
+          ? (popToNextHostParent(fiber),
+            5 === fiber.tag &&
+              null != fiber.stateNode &&
+              (nextHydratableInstance = fiber.stateNode))
+          : (popToNextHostParent(fiber), (isHydrating = !0)));
     default:
       unwindInterruptedWork(current, next),
         (next = workInProgress =
